### PR TITLE
fix: ensure `sender_id` has a balance is greater than amount

### PR DESCRIPTION
**Pull request by 0x3bfc**
# Description

The function `ft_transfer_call()` allows sending an arbitrary amount of tokens if `sender_id == receiver_id`. This PR adds an extra check to ensure that the sender_id has enough balance.

### CODE REVIEW

**Comment by Aurora-ClickUp** [comment]
Task linked: [CU-863g0gfem AUR-11 Infinite Amounts Can Be Transferred to One’s Self](https://app.clickup.com/t/863g0gfem)

File: engine-tests/src/tests/eth_connector.rs [hunk 1]
```diff
@@ -501,6 +501,46 @@ fn test_ft_transfer_call_without_message() {
     let balance = get_eth_on_near_balance(&master_account, CONTRACT_ACC, CONTRACT_ACC);
     assert_eq!(balance, DEPOSITED_FEE);
 
+    // should revert with `not enough balance` error when sending arbitray amount while sender_id = receiver_id
```
Comment (aleksuss) [inline engine-tests/src/tests/eth_connector.rs:504]:
```suggestion
    // should revert with `not enough balance` error when sending arbitrary amount while sender_id == receiver_id
```

File: engine-tests/src/tests/eth_connector.rs [hunk 2]
```diff
@@ -501,6 +501,46 @@ fn test_ft_transfer_call_without_message() {
...
+        1,
+    );
+
+    assert_execution_status_failure(
+        res.outcome().clone().status,
+        "ExecutionError(\"Smart contract panicked: ERR_NOT_ENOUGH_BALANCE\")",
+        "Expected failure in `ft_transfer_call` call, but call succeeded",
+    );
+
+    // should not revert with `not enough balance` error when sending arbitray amount while sender_id = receiver_id with amount < balance
```
Comment (aleksuss) [inline engine-tests/src/tests/eth_connector.rs:526]:
```suggestion
    // should not revert with `not enough balance` error when sending arbitrary amount while sender_id == receiver_id with amount < balance
```

**Review by aleksuss — APPROVED** [review]
Overall looks good

**Comment by joshuajbouw** [comment]
> Looks fine to me. Not sure why the CI didn't trigger on the last commit though

Happens from time to time. Github bug I think.

**Review by joshuajbouw — APPROVED** [review]

**Review by hskang9 — COMMENTED** [review]
LGTM

**Pull request merged**